src/main/webapp/3Dscript.js

```diff
@@ -11,15 +11,61 @@ let controls;
 let orientation;
 let trajectory;
 let pose;
+let poseLength;
```

Comment:
Just use `pose.length`, to avoid keeping multiple variables in sync and to reduce the global namespace clutter.

Reply:
that was the original plan, and im not  sure what was the exact technical reason pose.length didnt work, but the slide bar in the gui had errors whenever i tried to call pose.length directly (i think the error was the call to make the gui slider did not define pose), so in the spirit of fixing the issue, i had to put the length in another variable (declared in line 435.)

Reply:
Hmm I don't see how it would matter to the slider whether the length is a property of `pose` or its own variable. It's more likely the problem was something like `pose` not being initialized yet. What exactly happens if you just replace `poseLength` with `pose.length` in `makeGUI` and `hideOrientation`?